core/monitor/verify.go

```diff
@@ -141,12 +141,21 @@ func (m *Monitor) verifyMutations(muts []*ktpb.MutationProof, oldRoot, expectedN
 		}
 
 		// compute the new leaf
-		newLeaf, err := mutator.Mutate(oldLeaf, mut.GetMutation())
+		newValue, err := mutator.Mutate(oldLeaf, mut.GetMutation())
 		if err != nil {
 			glog.Infof("Mutation did not verify: %v", err)
 			errList = append(errList, ErrInvalidMutation)
 		}
 		newLeafnID := storage.NewNodeIDFromPrefixSuffix(index, storage.Suffix{}, m.mapHasher.BitLen())
+		newLeaf, err := entry.ToLeafValue(newValue)
+		if err != nil {
+			glog.Infof("Failed to serialize: %v", err)
+			errList = append(errList, err)
+		}
+
+		// BUG(gdbelvin): Proto serializations are not idempotent.
```

Comment:
Is this bug already present?  Or was it previously not an issue?

Reply:
This bug was already present. I'm just flagging it here so I don't forget it when I come back to this code base.

Reply:
OK

Reply:
OK sgtm